app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,82 @@
+class PostsController < ApplicationController
+  before_action :set_post, only: [:show, :edit, :update, :destroy]
+  before_action :fetch_blogs, only: [:new, :edit]
+  before_action :authenticate_user!
+  # GET /posts
+  # GET /posts.json
+  def index
+    @posts = Post.where(user_id: current_user.id)
+  end
+
+  # GET /posts/1
+  # GET /posts/1.json
+  def show
+  end
+
+  # GET /posts/new
+  def new
+    @post = Post.new
+  end
+
+  # GET /posts/1/edit
+  def edit
+  end
+
+  # POST /posts
+  # POST /posts.json
+  def create
+    @post = Post.new(post_params)
+    @post.user_id = current_user.id
+    @post.tag_list.add(params[:post][:tag_list])
+
+    respond_to do |format|
+      if @post.save
+        format.html { redirect_to @post, notice: 'Post was successfully created.' }
+        format.json { render :show, status: :created, location: @post }
+      else
+        format.html { render :new }
+        format.json { render json: @post.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /posts/1
+  # PATCH/PUT /posts/1.json
+  def update
+    respond_to do |format|
+      @post.tag_list.add(params[:post][:tag_list])
+      if @post.update(post_params)
+        format.html { redirect_to @post, notice: 'Post was successfully updated.' }
+        format.json { render :show, status: :ok, location: @post }
+      else
+        format.html { render :edit }
+        format.json { render json: @post.errors, status: :unprocessable_entity }
+      end
+    end
+  end
```

Comment:
Same deal here RE authorization—any user seems to be able to update any post for any blog regardless of whether it's theirs, albeit with a bit of fiddling to change the blog ID in the select field.

Reply:
Interestingly, when creating or updating a post without first selecting a blog, I get an exception page rather than a validation error message.

Reply:
these have been resolved too;